crates/iceberg-catalog/src/service/storage/az.rs

```diff
@@ -0,0 +1,520 @@
+        let table_id = uuid::Uuid::now_v7();
+        let test_location =
+            self.tabular_location(ns_id, TabularIdentUuid::Table(table_id)) + "/test_file";
+        let sanitized_ns_location = super::path_utils::sanitize_azdls_path(&namespace_location);
+        storage::validate_file_io(&file_io, &test_location)
+            .await
+            .map_err(|e| e.error.append_detail(format!("Profile: {self:?}")))?;
+
+        tracing::debug!(
+            "Successfully validated metadata writing, moving on to validating table config.",
+        );
+
+        self.validate_sas(credential, ns_id, table_id, test_location)
+            .await?;
+
+        tracing::debug!("Successfully validated table_config by using sas token, moving on to cleaning up by deleting contents of: '{}'.", sanitized_ns_location);
+
+        file_io
+            .remove_all(sanitized_ns_location)
+            .await
+            .map_err(|err| {
+                ErrorModel::internal(
+                    "Failed to delete test location.",
+                    "AzStorageDeletionError",
+                    Some(Box::new(err)),
+                )
+            })?;
+
+        tracing::debug!("Cleaned up");
+
+        Ok(())
+    }
+
+    /// Check if the profile can be updated with the other profile.
+    /// `key_prefix`, `region` and `bucket` must be the same.
+    /// We enforce this to avoid issues by accidentally changing the bucket or region
+    /// of a warehouse, after which all tables would not be accessible anymore.
+    /// Changing an endpoint might still result in an invalid profile, but we allow it.
+    ///
+    /// # Errors
+    /// Fails if the `bucket`, `region` or `key_prefix` is different.
+    pub fn can_be_updated_with(&self, other: &Self) -> Result<()> {
+        if self.filesystem != other.filesystem {
+            return Err(ErrorModel::bad_request(
+                "Storage Profile `filesystem` cannot be updated to prevent data loss.",
+                "InvalidFileSystem",
+                None,
+            )
+            .into());
+        }
+
+        if self.key_prefix != other.key_prefix {
+            return Err(ErrorModel::bad_request(
+                "Storage Profile `key_prefix` cannot be updated to prevent data loss.",
+                "InvalidKeyPrefix",
+                None,
+            )
+            .into());
+        }
+
+        Ok(())
+    }
+
+    #[must_use]
+    pub fn generate_catalog_config(&self, _: WarehouseIdent) -> CatalogConfig {
+        CatalogConfig {
+            defaults: HashMap::default(),
+            overrides: HashMap::default(),
+        }
+    }
+
+    #[must_use]
+    pub fn tabular_location(
+        &self,
+        namespace_id: NamespaceIdentUuid,
+        tabular_id: TabularIdentUuid,
+    ) -> String {
+        format!("{}{tabular_id}", self.namespace_location(namespace_id))
+    }
+
+    fn namespace_location(&self, namespace_id: NamespaceIdentUuid) -> String {
```

Comment:
Same here as for `tabular_location`, lets rename to `initial_`, ...

Reply:
renamed